### PR TITLE
tool for loading CommonJS modules lazily

### PR DESCRIPTION
**Pull request by vankop**
**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

no

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

Nothing.

Since webpack 5 support only Node 10 and higher this util function can be used for loading modules asynchronously 

I have checked code base using  regex `^(?!const)\s*[a-zA-Z\d]+\s*=\s*require\(.*\)` in `lib` folder for possible usage, maybe there are other places too

### CODE REVIEW

**Comment by vankop** [comment]
@sokra What is the best way to run benchmarks for `webpack@5`? `yarn run benchmark` looks like abandoned command or it must be updated =)

File: lib/util/lazyRequire.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+/* Fork from https://github.com/sindresorhus/import-lazy */
```
Comment (sokra) [inline lib/util/lazyRequire.js]: You can't copy code from somewhere and omit the copyright in the license.

**Comment by vankop** [comment]
> The performance suffer when going through a Proxy

Yep, I wanted to run benchmarks, but did not found out how to do it better `node benchmark/benchmark` or `yarn run benchmark` ? (second command does not work).

If this is a problem for speculative optimization, I can suggest only to force `get`

> Instead you could try to use memorization instead:

Idea of this approach was to use required modules as is, without any extra function calls: `new (getAssetParser())()` or `const AssetParser = getAssetParser(); new AssetParser()`

> the require("../util/lazyRequire")(require) construct is hard to understand for webpack. You won't be able to webpack webpack and some tools rely on that. That's a big blocker.

We can mark it with `PURE` annotation before or on fly, or there is another problem?

**Comment by sokra** [comment]
> We can mark it with `PURE` annotation before or on fly, or there is another problem?

The problem is that `require` is used without calling it.

> Idea of this approach was to use required modules as is, without any extra function calls: `new (getAssetParser())()` or `const AssetParser = getAssetParser(); new AssetParser()`

Yes it would look better, but that doesn't really matter as much as performance matters.

**Comment by sokra** [comment]
Thanks

**Pull request merged**